.husky/pre-commit

```diff
@@ -1,4 +1,7 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-pnpm lint-staged --quiet
+LATEST_HASH=$(git rev-parse --verify HEAD)
+
+# Run commands only in packages with changed files compared to previous commit and on their dependants in parallel
+pnpm -r --filter="...[$LATEST_HASH]" --parallel "/^format$|^lint$|^typecheck$/"
```

Comment:
Nice, this is faster? 

I'm a little skeptical of pre-commit jobs. I've tried using them in the past and it gets annoying having a delay post-commit and I end up --no-verifying relying on CI anyway

Reply:
Not a big fan of it too but it's nice to keep the option for now. We can remove it down the road if it turns out too big a of a pain.

running with parallel recursive pnpm is faster right now, might become a bit slower than lint-staged down the road. But on the plus side, we can run tsc, which imho usually gives more info than eslint.